fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -602,10 +585,16 @@ protected void replaceResourceWithStream(final FedoraResource resource,
     }
 
     protected void patchResourcewithSparql(final FedoraResource resource,
-                                           final String requestBody,
-                                           final RdfStream resourceTriples)
-            throws MalformedRdfException, AccessDeniedException {
-        resource.updateProperties(translator(), requestBody, resourceTriples);
+            final String requestBody,
+            final RdfStream resourceTriples)
+                    throws MalformedRdfException, AccessDeniedException {
+        if (resource instanceof NonRdfSourceDescription) {
+            // update the description instead
```

Comment:
We need to make sure that if we are going with the design in this ticket, that it reconcile with:
https://jira.duraspace.org/browse/FCREPO-1540 and
https://jira.duraspace.org/browse/FCREPO-1590